strr-platform-web/app/pages/platform/application.vue

```diff
@@ -125,16 +126,65 @@ const setPreviousStep = () => {
   }
 }
 
-// something like this? need to discuss options
-// how can we set loading state on the submit and pay button?
-// i wonder if tracking the steps in a store would be useful
-const handlePlatformSubmit = () => {
-  // validate each step
-  // if invalid step, set prop on Review component to highlight invalid fields/section ?
-  // show alert with some error text ?
+// need to cleanup the setButtonControl somehow
+const handlePlatformSubmit = async () => {
+  try {
+    // set buttons to loading state
+    setButtonControl({
+      leftButtons: [
+        {
+          action: setPreviousStep,
+          icon: 'i-mdi-chevron-left',
+          label: t('btn.back'),
+          variant: 'outline',
+          disabled: true
+        },
+        {
+          action: handlePlatformSubmit,
+          icon: 'i-mdi-chevron-right',
+          label: t('btn.submitAndPay'),
+          trailing: true,
+          loading: true
+        }
+      ],
+      rightButtons: []
+    })
+
+    // something like this but cleaner
```

Comment:
```suggestion
    // TODO: something like this but cleaner
```
So it shows up in the TODO tracking